src/mtev_reverse_socket.c

```diff
@@ -387,6 +412,7 @@ mtev_reverse_socket_channel_handler(eventer_t e, int mask, void *closure,
     eventer_remove_fd(e->fd);
     CHANNEL.pair[0] = CHANNEL.pair[1] = -1;
     mtev_reverse_socket_channel_shutdown(cct->parent, cct->channel_id, e);
+    mtev_reverse_socket_deref(cct->parent);
```

Comment:
We are setting channel pairs to -1 right before the call..... we only deref in the function if those are not -1, so we won't deref in the function here.